org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/SwitchStatement.java

```diff
@@ -1023,7 +1023,9 @@ char[] typeSwitchSignature(TypeBinding exprType) {
 		char[] arg1 = switch (exprType.id) {
 			case TypeIds.T_JavaLangLong, TypeIds.T_JavaLangFloat, TypeIds.T_JavaLangDouble, TypeIds.T_JavaLangBoolean,
 				TypeIds.T_JavaLangByte, TypeIds.T_JavaLangShort, TypeIds.T_JavaLangInteger, TypeIds.T_JavaLangCharacter->
-				exprType.signature();
+				this.isPrimitiveSwitch
+				? exprType.signature()
+				: "Ljava/lang/Object;".toCharArray(); //$NON-NLS-1$
```

Comment:
there are two more locations below this block which still use `exprType.signature()` unconditionally. Shouldn't those have the same guard, too?

Reply:
... or, simpler still, make an early return if `!this.isPrimitiveSwitch`.

Reply:
yep, will take a look - will wait for an I build first to confirm the fix